src/inference/src/dev/core_impl.cpp

```diff
@@ -1008,6 +1018,7 @@ std::shared_ptr<const ov::Model> ov::CoreImpl::apply_auto_batching(const std::sh
 
         deviceNameWithBatchSize = deviceName.substr(pos + 1);
         deviceNameWithoutBatch = ov::DeviceIDParser::get_batch_device(deviceNameWithBatchSize);
```

Comment:
should we insert the logic directly to `ov::DeviceIDParser::get_batch_device`?
It's responsible for parsing and, hence, for validation